pango/src/layout.rs

```diff
@@ -0,0 +1,52 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::translate::*;
+
+use crate::LayoutLine;
+
+/// The result of [`LayoutLine::x_to_index`].
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct HitPosition {
+    /// The UTF-8 byte offset of the grapheme closest to the position.
+    ///
+    /// This position is relative to the start of the line, not the start
+    /// of the [`Layout`]'s text.
+    ///
+    /// [`Layout`]: crate::Layout
+    pub index: i32,
+    /// The codepoint within the grapheme of the position.
+    ///
+    /// This will always be either `0`, or the number of `char`s (*not bytes!*)
+    /// in the grapheme. This represents whether the user clicked near the start
+    /// of the grapheme or near the end; this is important for things like
+    /// resolving cursor positions.
+    pub trailing: i32,
+    /// Whether or not the position was within the bounds of the line.
+    ///
+    /// If this is `false`, then `index` and `trailing` will always resolve
+    /// to either the very first or the very last position in the line; this
+    /// behaviour is dependent on the line's resolved writing direction.
+    pub is_inside: bool,
+}
+
+impl LayoutLine {
+    /// Return the [`HitPosition`] corresponding to the provided `x_pos` in this line.
```

Comment:
So like said below, please remove this doc comment. Documentation will be generated automatically like for the rest of the codebase.